src/seedu/addressbook/data/person/Address.java

```diff
@@ -25,7 +29,15 @@ public Address(String address, boolean isPrivate) throws IllegalValueException {
         if (!isValidAddress(address)) {
             throw new IllegalValueException(MESSAGE_ADDRESS_CONSTRAINTS);
```

Comment:
You also need to delegate validation of the address parts to the new classes you have created. that is, for example, Block should check  the validity of the Block's format.